### PR TITLE
DSPDisassembler: various code cleanups

### PR DESCRIPTION
**Pull request by ligfx**

### CODE REVIEW

File: Source/Core/Core/DSP/DSPDisassembler.cpp [hunk 1]
```diff
@@ -25,49 +25,14 @@ DSPDisassembler::DSPDisassembler(const AssemblerSettings& settings) : settings_(
 {
 }
 
-DSPDisassembler::~DSPDisassembler()
+bool DSPDisassembler::Disassemble(const std::vector<u16>& code, std::string& text)
```
Comment (lioncash) [inline Source/Core/Core/DSP/DSPDisassembler.cpp]: This only ever returns true now. Should this not check if the individual opcodes successfully disassemble and return false if one doesn't?

Comment (ligfx) [inline Source/Core/Core/DSP/DSPDisassembler.cpp]: Good catch! Yeah, it probably should, huh?

**Comment by shuffle2** [comment]
@dolphin-emu-bot rebuild

**Comment by shuffle2** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/DSP/DSPDisassembler.cpp [hunk 2]
```diff
@@ -25,49 +25,15 @@ DSPDisassembler::DSPDisassembler(const AssemblerSettings& settings) : settings_(
...
 {
-  // Some old code for logging unknown ops.
-  std::string filename = File::GetUserPath(D_DUMPDSP_IDX) + "UnkOps.txt";
-  std::ofstream uo(filename);
-  if (!uo)
-    return;
-
-  int count = 0;
-  for (const auto& entry : unk_opcodes)
+  for (u16 pc = 0; pc < code.size();)
```
Comment (shuffle2) [inline Source/Core/Core/DSP/DSPDisassembler.cpp]: kinda overkill, but might want to check `code.size()` won't cause inf loop here?

Comment (ligfx) [inline Source/Core/Core/DSP/DSPDisassembler.cpp]: Are you thinking that `DisassembleOpcode` might not increment `pc`, or that `code.size()` might be larger than what can be represented by a `u16`?

Comment (shuffle2) [inline Source/Core/Core/DSP/DSPDisassembler.cpp]: The latter.

Comment (ligfx) [inline Source/Core/Core/DSP/DSPDisassembler.cpp]: Done.

**Pull request merged**